pkg/cloud/services/networking/securitygroups.go

```diff
@@ -295,6 +297,9 @@ func (s *Service) exists(groupID string) (bool, error) {
 	}
 	allGroups, err := groups.ExtractGroups(allPages)
 	if err != nil {
+		if capierrors.IsNotFound(err) {
```

Comment:
Does ExtractGroups really throw these errors? I would have guessed they are returned by "network" calls like Get/List etc?

Reply:
um... I didn't test this, so I will remove this check 
after further test , we can add through another PR